### ISSUE TITLE
linter: false positive on react/capitalized-calls with Date.UTC

### ISSUE THREAD

**Issue by chambo-e** [oxc-project/oxc#26678]
### What version of Oxlint are you using?

1.83.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint'

export default defineConfig({
  plugins: ['react'],
  categories: {
    correctness: 'off',
  },
  rules: {
    'react/capitalized-calls': 'error',
  }
})
```

### What happened?

Hello, 

I don't know if it's the intended behavior or not

react/capitalized-calls throws an error on Date.UTC calls

```tsx
export const Sample = () => {
    const now = new Date()
    // here Capitalized function called without JSX
    const startDate = new Date(Date.UTC(now.getUTCFullYear(), now.getUTCMonth(), 1))

    return <div>{startDate.toString()}</div>
}
```

Thanks !
